### PR TITLE
Carbon(graphite) proxy service

### PR DESCRIPTION
**Pull request by antmat**
Service adds ability to send metrics to carbon and acts as a proxy.
Protocol provides:
send_one([metric_name, metric_value, optional<metric_timestamp>])
send_bulk([[metric_name, metric_value, optional<metric_timestamp>], ...])
Service acts as a proxy. It aggregates data for some period (flush_interval_ms in config, default 1000ms) and resends data in plain text protocol to carbon in one chunk.
On connection/transmission errors service discards metrics and writes a warning to log.

### CODE REVIEW

**Comment by 3Hren** [comment]
LGTM, except some minor notices, which are told personally.

**Comment by noxiouz** [comment]
LGTM

**Comment by Kobolog** [comment]
I'm going to review it. Don't merge.

**Comment by Kobolog** [comment]
:+1:

**Comment by Kobolog** [comment]
But next time keep PRs to the point. This one includes some code reorganization stuff not related to the plugin, for example.

**Comment by 3Hren** [comment]
Status: waiting for results from manual acceptance tests. Will be merged if ok.

**Comment by 3Hren** [comment]
Please, rebase duplicate commits.

File: docker/docker_client.cpp [hunk 1]
```diff
@@ -30,6 +30,18 @@
...
+namespace {
+int
+curl_init() {
+    if(curl_global_init(CURL_GLOBAL_ALL) != CURLE_OK) {
+        std::cout << "Curl initialization failed. Aborting execution." << std::endl;
+        std::terminate();
+    }
+    return 42;
+}
+int curl_init_placeholder = curl_init();
```
Comment (3Hren) [inline docker/docker_client.cpp]: Write several words about why this magic is necessary.

**Pull request merged**